utils/include/detray/simulation/event_writer.hpp

```diff
@@ -89,8 +89,7 @@ struct event_writer : actor {
         auto& stepping = propagation._stepping;
 
         // triggered only for sensitive surfaces
-        if (navigation.is_on_module() &&
```

Comment:
Why is this not possible for the ```propagator_sync```? It will lead to much less branching code in the navigator

Reply:
If we just use `is_on_module`, we will synchronize a thread on passive and a thread on sensitive, which can be inefficient.
Possibly it could be useful in certain cases, so this might need to be generalized based on where we want to apply the synchronization.